cirq/protocols/approximate_equality_protocol.py

```diff
@@ -94,6 +95,11 @@ def approx_eq(val: Any, other: Any, *, atol: Union[int, float] = 1e-8) -> bool:
     if isinstance(val, str):
         return val == other
 
+    if isinstance(val, sympy.core.mul.Mul) or isinstance(
```

Comment:
use sympy.Basic instead Mul is way too restrictive

We should also add a test case maybe directly against other expressions e.g.
```
cirq.approx_eq(t+1.51+s, t+1.50+s, atol=0.1)
```
Would evaluate to False with this code.

Reply:
Changed to Basic

Added the direct test case, although since the tolerance was right on the threshold, I decided to add two tests, once with tolerate 2.0, one with tolerance 0.5

Reply:
Why was the tolerance on the threshold? 1.51-1.50 = 0.01 < 0.1

Reply:
It was a mistake. Changed to 0.2 for the other tests.